### PR TITLE
fix(radio-group): replaced label with content

### PR DESCRIPTION
**Pull request by bmdalex**
# Radio group

This PR contains:
- replacing `label` prop with `content`
- making `content` accept react nodes as value

### TODO

- [x] Conformance test
- [x] Minimal doc site example
- [ ] Stardust base theme
- [ ] Teams Light theme
- [x] Teams Dark theme
- [ ] Teams Contrast theme
- [ ] Confirm RTL usage
- [ ] [W3 accessibility](https://www.w3.org/standards/webdesign/accessibility) check
- [ ] [Stardust accessibility](https://github.com/stardust-ui/accessibility) check
- [ ] Update glossary props table
- [x] Update the CHANGELOG.md

### CODE REVIEW

File: src/components/RadioGroup/RadioGroupItem.tsx
Comment (kuzhelov) [inline src/components/RadioGroup/RadioGroupItem.tsx]: `shorthand` test should be added for this prop (`implementsShorthandProp`) then

Comment (kuzhelov) [reply src/components/RadioGroup/RadioGroupItem.tsx]: but, in fact, this is not a shorthand.. seems that propType name (`contentShorthand`) is misleading - please, ignore the comment about tests necessary

File: src/components/RadioGroup/RadioGroupItem.tsx
Comment (kuzhelov) [inline src/components/RadioGroup/RadioGroupItem.tsx]: a bit worrying about semantics behind this nesting here. Do we really want the content (that is quite generic by its nature) to be always a children of `Label` (and, thus, to always reflect its styles and, what is more important, behavioural and accessibility aspects - for _any arbitrary value of `content`_ that will be passed in)?

Comment (bmdalex) [reply src/components/RadioGroup/RadioGroupItem.tsx]: I'm not changing the implementation as you can see, just the name of the prop (I should have kept the name of the styles slot to `label`, will revert that part).

I think this is a legitimate concern, but the nature of the radio button dictates to us that we need an `icon` with some `content` after it; maybe we should not use `Label` to wrap these, any suggestions?

@kuzhelov @mnajdova

File: src/components/RadioGroup/RadioGroupItem.tsx
Comment (bmdalex) [reply src/components/RadioGroup/RadioGroupItem.tsx]: any changes you'd suggest?

Comment (kuzhelov) [reply src/components/RadioGroup/RadioGroupItem.tsx]: no-no, nothing - let me close this one, sorry for misleading you

File: src/components/RadioGroup/RadioGroupItem.tsx
Comment (kuzhelov) [reply src/components/RadioGroup/RadioGroupItem.tsx]: yes, I would suggest exactly the same path as you've done - specifically, to not wrap Icon with Label (especially if we would take into account where Icon is used - it is used to 'fill' the circle area of radio input). Then it would be much easier to decouple all these parts from each other.

Not sure if we should do it in this PR or create a dedicated one as a close follow-up, but the most important thing would be, actually, if we all agree on that proposal :)

Comment (bmdalex) [reply src/components/RadioGroup/RadioGroupItem.tsx]: I'd rather do it in another PR; I'll discuss with @jurokapsiar as well to understand what are the concerns (if any) and proceed with removing Label

File: src/components/Input/Input.tsx
Comment (mnajdova) [inline src/components/Input/Input.tsx]: Did we fix here the input prop not to be propagated as a prop to the ElementType?

File: src/components/RadioGroup/RadioGroupItem.tsx
Comment (kuzhelov) [inline src/components/RadioGroup/RadioGroupItem.tsx]: not sure whether we would like to expose it as a prop - most probably this aspect will be handled by accessibility behavior. Could we omit it for this PR, or there is specific urgent need for this functionality to be introduced?

File: src/components/Input/Input.tsx
Comment (kuzhelov) [inline src/components/Input/Input.tsx]: have we agreed to have this `input` prop for the `Input` element (like as it is in SUIR)? We've had some concerns - specifically, the fact that it makes it non-intuitive for the client what `input` prop of `Input` component is about. Maybe we should leave it for a dedicated PR?

**Review by kuzhelov — COMMENTED** [review]
generally agree with the changes made, but it seems that there is more changes than necessary to address the goal of this PR. Would rather suggest to do it step-by-step - or to discuss those additional concepts introduced, as those might raise some concerns (have expressed them in comments)

File: src/components/Input/Input.tsx
Comment (bmdalex) [reply src/components/Input/Input.tsx]: then we won't be able to achieve functionality of `RadioGroup` items that contain inputs.. Marija told we we're expected to merge a dedicated PR as well; I'm ok with removing it from here but the examples with `input` in this PR won't work correctly

@kuzhelov @mnajdova

File: src/components/RadioGroup/RadioGroupItem.tsx
Comment (bmdalex) [reply src/components/RadioGroup/RadioGroupItem.tsx]: this is solving the bug I was telling you about.

#### Repro:
- create a `RadioGroup` with a `RadioGroupItem` with input (as in the examples of this PR);
- click on input;

#### Expected:
- item is checked and focus stays on input (see `Teams` `Add people` dialog in a group chat)

#### Actual:
- the radio item steals the focus;

This is the reason we're using this `shouldFocus` prop in order to set focus only when it's coming from keyboard; 

Unfortunately accessibility functionality spilled in our component logic and this became necessary now to do it like this. Alternative (and probably better approach) would be to rethink the whole accessibility part

What do you think?
@kuzhelov @mnajdova @jurokapsiar

Comment (bmdalex) [reply src/components/RadioGroup/RadioGroupItem.tsx]: fyi @smykhailov

Comment (bmdalex) [reply src/components/RadioGroup/RadioGroupItem.tsx]: adding this prop is critical to enable keyboard navigation for this component ASAP; I created an issue #306  for implementing it properly once @smykhailov 's work in #256 is merged

fyi @mnajdova @kuzhelov @jurokapsiar 

It'd be great to approve this ASAP so we can unblock customers

File: docs/src/examples/components/RadioGroup/Types/RadioGroupExample.shorthand.tsx
Comment (mnajdova) [inline docs/src/examples/components/RadioGroup/Types/RadioGroupExample.shorthand.tsx]: I am bit confuse here... Can we still use the input shorthand, now that we don't have it in the Input component?

Comment (mnajdova) [reply docs/src/examples/components/RadioGroup/Types/RadioGroupExample.shorthand.tsx]: Watch out, this definition that you have is different then having 
```
label: {
  content: (
    <Text>
      Choose your own{' '}
      <Input input={{ tabIndex: this.state.inputTabIndex }} inline placeholder="flavour" />
     </Text>
  )
}
``` 
With your definition you are just having this component rendered, so any default or overridden props that you have for the label component will not be applied.

File: src/components/RadioGroup/RadioGroupItem.tsx
Comment (mnajdova) [reply src/components/RadioGroup/RadioGroupItem.tsx]: I agree with @Bugaa92 on this. We can come back and refactor this, otherwise this basic functionality will be blocked for at least a week from now unfortunately.

File: CHANGELOG.md
Comment (mnajdova) [inline CHANGELOG.md]: Please update this, as we are not really replacing the label after all :)

File: docs/src/examples/components/RadioGroup/Types/RadioGroupExample.shorthand.tsx
Comment (bmdalex) [reply docs/src/examples/components/RadioGroup/Types/RadioGroupExample.shorthand.tsx]: > I am bit confuse here... Can we still use the input shorthand, now that we don't have it in the Input component?

no, this will work correctly once we fix the input shorthand, but it does no harm now, just key nav is not working correctly for the case when we tab from an item to the next component

> With your definition you are just having this component rendered, so any default or overridden props that you have for the label component will not be applied.

I don't get it, that's what I want.. the label slot is already inside `Label` component

Comment (mnajdova) [reply docs/src/examples/components/RadioGroup/Types/RadioGroupExample.shorthand.tsx]: > I don't get it, that's what I want.. the label slot is already inside `Label` component

Sorry, my mistake, I thought this contain `Label.create(label ...` It is defined as it should.

Comment (mnajdova) [reply docs/src/examples/components/RadioGroup/Types/RadioGroupExample.shorthand.tsx]: Let's get rid of the input shorthand, and we will add it in the Input's PR

**Review by kuzhelov — APPROVED** [review]

File: CHANGELOG.md
Comment (bmdalex) [reply CHANGELOG.md]: done

File: src/components/Input/Input.tsx
Comment (bmdalex) [reply src/components/Input/Input.tsx]: we will do it in another PR

File: docs/src/examples/components/RadioGroup/Types/RadioGroupExample.shorthand.tsx
Comment (bmdalex) [reply docs/src/examples/components/RadioGroup/Types/RadioGroupExample.shorthand.tsx]: done

**Pull request merged**
